integrations/Matomo/util.js

```diff
@@ -0,0 +1,291 @@
+import get from "get-value";
+import each from "@ndhoule/each";
+import logger from "../../utils/logUtil";
+import { getHashFromArray } from "../utils/commonUtils";
+
+/** If any event name matches with the goals list provided by the dashboard
+ * we will call the trackGoal with the id provided in the mapping.
+ */
+const goalIdMapping = (event, goalListTo) => {
+  each((val, key) => {
+    if (key === event.toLowerCase()) {
+      // val is the goalId provided in the dashboard
+      window._paq.push(["trackGoal", val]);
+    }
+  }, goalListTo);
+};
+
+/** Mapping Standard Events 
+ If any event name matches with the standard events list provided in the dashboard
+ */
+const standardEventsMapping = (event, standardTo, properties) => {
+  each((val, key) => {
+    if (key === event.toLowerCase()) {
+      let url;
+      let linkType;
+      let keyword;
+      let category;
+      let resultsCount;
+      let contentInteraction;
+      let contentName;
+      let contentPiece;
+      let contentTarget;
+
+      switch (val) {
+        case "trackLink":
+          url = get(properties, "url");
+          linkType = get(properties, "linkType");
+          window._paq.push(["trackLink", url, linkType]);
+          break;
+
+        case "trackSiteSearch":
+          keyword = get(properties, "keyword");
+          category = get(properties, "category");
+          resultsCount = get(properties, "resultsCount");
+          window._paq.push([
+            "trackSiteSearch",
+            keyword,
+            category,
+            resultsCount,
+          ]);
+          break;
+
+        case "ping":
+          window._paq.push(["ping"]);
+          break;
+
+        case "trackContentImpressionsWithinNode":
+          window._paq.push(["trackContentImpressionsWithinNode", document]);
```

Comment:
We should call this event for the given domNode. Not for the entire document.

Reply:
Done